clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1198,13 +1207,40 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
 
     LastParamIndex = Params.size();
     Params.push_back(NewParam);
+    BufferLocationMD.push_back(LocationID);
+  }
+
+  // Obtain an integer value stored in a template parameter of buffer_location
+  // property to pass it to buffer_location kernel attribute
+  size_t handleBufferLocationProperty(QualType FieldTy) {
+    const auto *AccTy =
+        cast<ClassTemplateSpecializationDecl>(FieldTy->getAsRecordDecl());
+
+    // TODO: when SYCL headers' part is ready - replace this 'if' with an assert
+    if (AccTy->getTemplateArgs().size() < 6)
+      return -1;
+
+    // TODO: at this point of time it's unclear, what representation in LLVM IR
+    // is going to be for other compile time known accessor properties, hence
+    // it's not clear, how handle them in SemaSYCL. But in general property_list
+    // is a parameter pack and shall be handled appropriately.
+    const auto Prop = cast<TemplateArgument>(AccTy->getTemplateArgs()[5]);
```

Comment:
This assumption of the location of the layout of the type needs to be diagnosed as well.  Probably in the field checker.